crates/ntx-builder/src/builder/mod.rs

```diff
@@ -62,130 +85,182 @@ impl NetworkTransactionBuilder {
+                                }
+                            }
+                        }
+                        Some(Err(err)) => {
+                            if err.is_panic() {
+                                tracing::error!(err = ?err, "actor join set panicked");
+                                // TODO: exit?
+                            } else {
+                                tracing::error!(err = ?err, "actor join set failed");
+                                // TODO: exit?
                             }
-                            state.candidate_failed(candidate);
                         },
-                        // Unexpected error occurred.
-                        Err(err) => {
-                            tracing::warn!(err=err.as_report(), "network transaction panicked");
-                            state.candidate_failed(candidate);
+                        None => {
+                            tracing::warn!("actor join set is empty");
+                            tokio::time::sleep(Duration::from_secs(1)).await;
                         }
                     }
-                }
+                },
+                // Handle mempool events.
+                event = mempool_events.try_next() => {
+                    let event = event
+                        .context("mempool event stream ended")?
+                        .context("mempool event stream failed")?;
+
+                    self.handle_mempool_event(
+                        &event,
+                        &config,
+                    );
+                },
             }
         }
     }
-}
 
-/// A wrapper arounnd tokio's [`JoinSet`](tokio::task::JoinSet) which returns pending instead of
-/// [`None`] if its empty.
-///
-/// This makes it much more convenient to use in a `select!`.
-struct JoinSet<T>(tokio::task::JoinSet<T>);
-
-impl<T> JoinSet<T>
-where
-    T: 'static,
-{
-    fn new() -> Self {
-        Self(tokio::task::JoinSet::new())
+    fn handle_mempool_event(
+        &mut self,
+        event: &MempoolEvent,
+        account_actor_config: &AccountActorConfig,
+    ) {
+        match &event {
+            // Broadcast to affected actors.
+            MempoolEvent::TransactionAdded { account_delta, network_notes, .. } => {
+                // Find affected accounts.
+                let affected_accounts =
+                    Self::find_affected_accounts(account_delta.as_ref(), network_notes);
+
+                for account_prefix in affected_accounts {
+                    // Retrieve or create the actor.
+                    if let Some(actor_handle) = self.actor_registry.get(&account_prefix) {
+                        Self::send_event(actor_handle, event.clone());
+                    } else {
+                        // Try creating the actor.
+                        let handle = self.spawn_actor(account_prefix, account_actor_config);
+                        Self::send_event(&handle, event.clone());
```

Comment:
imo this is easier to reason about if we:

- Ensure that _all_ committed network accounts have a loaded actor which can be done at startup.
- We restart actors if they crash and we think it is sensible, maybe with three retries/rate-limit the reset.

We then need to handle actor spawning here for `Account::new` type transactions, and also removing/deleting the actor if the tx gets reverted.

Reply:
The latter _must_ be done separately from the store loading style actor. This is because there is no account in the store - the account data is only from the transaction event and you therefore need two methods to spawn an actor.

Reply:
> Ensure that all committed network accounts have a loaded actor which can be done at startup.

Could you elaborate on how to do that? Right now I still create initial actor set via
```rust
let notes = store.get_unconsumed_network_notes().await?;
```